src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1658,14 +1658,14 @@ fn local_fingerprints_deps(
         local.push(LocalFingerprint::RerunIfChanged { output, paths });
     }
 
-    local.extend(
-        deps.rerun_if_env_changed
-            .iter()
-            .map(|var| LocalFingerprint::RerunIfEnvChanged {
-                var: var.clone(),
-                val: env::var(var).ok(),
-            }),
-    );
+    local.extend(deps.rerun_if_env_changed.iter().map(|var| {
+        #[allow(clippy::disallowed_methods)]
+        let val = env::var(var).ok();
```

Comment:
This should be read from config. I'll change this if people agree on.

Reply:
imo I feel like this one needs a good write up explaining why we should go in either direction